src/hidimstat/dcrt.py

```diff
@@ -1,307 +1,375 @@
+                    ),
+                    params_lasso_distillation_y=(
+                        self.params_lasso_distillation_y
+                        if self.params_lasso_distillation_y is not None
+                        else self.params_lasso_screening
+                    ),
+                    fit_y=self.fit_y,
+                    n_jobs=1,  # the function is already called in parallel
+                    random_state=self.random_state,
+                )
+                for idx in selection_set
             )
-            for idx in selection_set
+        elif self.statistic == "random_forest":
+            # For distillation of X use least_square loss
+            results = Parallel(self.n_jobs, verbose=self.joblib_verbose)(
+                delayed(_rf_distillation)(
+                    X_,
+                    y_,
+                    idx,
+                    sigma_X=self.sigma_X,
+                    n_tree=self.n_tree,
+                    problem_type=self.problem_type,
+                    n_jobs=1,  # the function is already called in parallel
+                    random_state=self.random_state,
+                    params_lasso_distillation_x=(
+                        self.params_lasso_distillation_x
+                        if self.params_lasso_distillation_y is not None
+                        else self.params_lasso_screening
+                    ),
+                )
+                for idx in selection_set
+            )
+        else:
+            raise ValueError(f"{self.statistic} statistic is not supported.")
+        # contatenate result
+        self.selection_features = np.ones((n_features,), dtype=bool)
+        self.selection_features[self.non_selection] = 0
+        self.X_residual = np.array([result[0] for result in results])
+        self.sigma2 = np.array([result[1] for result in results])
+        self.y_residual = np.array([result[2] for result in results])
+        self.clf_x_residual = np.array([result[3] for result in results])
+        self.clf_y_residual = np.array([result[4] for result in results])
+        return self
+
+    def importance(
+        self,
+        fdr=0.1,
+        fdr_control="bhq",
+        reshaping_function=None,
+        scaled_statistics=False,
+    ):
+        """
+        Calculate p-values and identify significant features using the dCRT test statistics.
+        This function processes the results from dCRT to identify statistically significant
+        features while controlling for false discoveries. It assumes test statistics follow
+        a Gaussian distribution.
+
+        Parameters
+        ----------
+        fdr : float, default=0.1
+            Target false discovery rate level (0 < fdr < 1)
+        fdr_control : {'bhq', 'bhy', 'ebh'}, default='bhq'
+            Method for FDR control:
+            - 'bhq': Benjamini-Hochberg procedure
+            - 'bhy': Benjamini-Hochberg-Yekutieli procedure
+            - 'ebh': e-BH procedure
```

Comment:
Maybe it could be specified that this is for multiple testing but that we also return uncorrected p-values.

Reply:
I add the precision. 
However, this part will be probably remove because their is no meaning of doing it (see issue #212 )